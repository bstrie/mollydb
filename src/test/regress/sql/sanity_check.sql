VACUUM;

--
-- sanity check, if we don't have indices the test will take years to
-- complete.  But skip TOAST relations (since they will have varying
-- names depending on the current OID counter) as well as temp tables
-- of other backends (to avoid timing-dependent behavior).
--

-- temporarily disable fancy output, so catalog changes create less diff noise
\a\t

SELECT relname, relhasindex
   FROM mdb_class c LEFT JOIN mdb_namespace n ON n.oid = relnamespace
   WHERE relkind = 'r' AND (nspname ~ '^mdb_temp_') IS NOT TRUE
   ORDER BY relname;

-- restore normal output mode
\a\t

--
-- another sanity check: every system catalog that has OIDs should have
-- a unique index on OID.  This ensures that the OIDs will be unique,
-- even after the OID counter wraps around.
-- We exclude non-system tables from the check by looking at nspname.
--
SELECT relname, nspname
FROM mdb_class c LEFT JOIN mdb_namespace n ON n.oid = relnamespace
WHERE relhasoids
    AND ((nspname ~ '^mdb_') IS NOT FALSE)
    AND NOT EXISTS (SELECT 1 FROM mdb_index i WHERE indrelid = c.oid
                    AND indkey[0] = -2 AND indnatts = 1
                    AND indisunique AND indimmediate);
