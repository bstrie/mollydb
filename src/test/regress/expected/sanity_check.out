VACUUM;
--
-- sanity check, if we don't have indices the test will take years to
-- complete.  But skip TOAST relations (since they will have varying
-- names depending on the current OID counter) as well as temp tables
-- of other backends (to avoid timing-dependent behavior).
--
-- temporarily disable fancy output, so catalog changes create less diff noise
\a\t
SELECT relname, relhasindex
   FROM mdb_class c LEFT JOIN mdb_namespace n ON n.oid = relnamespace
   WHERE relkind = 'r' AND (nspname ~ '^mdb_temp_') IS NOT TRUE
   ORDER BY relname;
a|f
a_star|f
abstime_tbl|f
aggtest|f
array_index_op_test|t
array_op_test|f
b|f
b_star|f
box_tbl|f
bprime|f
bt_f8_heap|t
bt_i4_heap|t
bt_name_heap|t
bt_txt_heap|t
c|f
c_star|f
char_tbl|f
check2_tbl|f
check_tbl|f
circle_tbl|t
city|f
copy_tbl|f
d|f
d_star|f
date_tbl|f
default_tbl|f
defaultexpr_tbl|f
dept|f
dupindexcols|t
e_star|f
emp|f
equipment_r|f
f_star|f
fast_emp4000|t
float4_tbl|f
float8_tbl|f
func_index_heap|t
hash_f8_heap|t
hash_i4_heap|t
hash_name_heap|t
hash_txt_heap|t
hobbies_r|f
ihighway|t
inet_tbl|f
inhf|f
inhx|t
insert_tbl|f
int2_tbl|f
int4_tbl|f
int8_tbl|f
interval_tbl|f
iportaltest|f
kd_point_tbl|t
line_tbl|f
log_table|f
lseg_tbl|f
main_table|f
money_data|f
num_data|f
num_exp_add|t
num_exp_div|t
num_exp_ln|t
num_exp_log10|t
num_exp_mul|t
num_exp_power_10_ln|t
num_exp_sqrt|t
num_exp_sub|t
num_input_test|f
num_result|f
onek|t
onek2|t
path_tbl|f
person|f
mdb_aggregate|t
mdb_am|t
mdb_amop|t
mdb_amproc|t
mdb_attrdef|t
mdb_attribute|t
mdb_auth_members|t
mdb_authid|t
mdb_cast|t
mdb_class|t
mdb_collation|t
mdb_constraint|t
mdb_conversion|t
mdb_database|t
mdb_db_role_setting|t
mdb_default_acl|t
mdb_depend|t
mdb_description|t
mdb_enum|t
mdb_event_trigger|t
mdb_extension|t
mdb_foreign_data_wrapper|t
mdb_foreign_server|t
mdb_foreign_table|t
mdb_index|t
mdb_inherits|t
mdb_init_privs|t
mdb_language|t
mdb_largeobject|t
mdb_largeobject_metadata|t
mdb_namespace|t
mdb_opclass|t
mdb_operator|t
mdb_opfamily|t
mdb_pltemplate|t
mdb_policy|t
mdb_proc|t
mdb_range|t
mdb_replication_origin|t
mdb_rewrite|t
mdb_seclabel|t
mdb_shdepend|t
mdb_shdescription|t
mdb_shseclabel|t
mdb_statistic|t
mdb_tablespace|t
mdb_transform|t
mdb_trigger|t
mdb_ts_config|t
mdb_ts_config_map|t
mdb_ts_dict|t
mdb_ts_parser|t
mdb_ts_template|t
mdb_type|t
mdb_user_mapping|t
point_tbl|t
polygon_tbl|t
quad_point_tbl|t
radix_text_tbl|t
ramp|f
real_city|f
reltime_tbl|f
road|t
shighway|t
slow_emp4000|f
sql_features|f
sql_implementation_info|f
sql_languages|f
sql_packages|f
sql_parts|f
sql_sizing|f
sql_sizing_profiles|f
stud_emp|f
student|f
tenk1|t
tenk2|t
test_range_excl|t
test_range_gist|t
test_range_spgist|t
test_tsvector|f
testjsonb|f
text_tbl|f
time_tbl|f
timestamp_tbl|f
timestamptz_tbl|f
timetz_tbl|f
tinterval_tbl|f
varchar_tbl|f
-- restore normal output mode
\a\t
--
-- another sanity check: every system catalog that has OIDs should have
-- a unique index on OID.  This ensures that the OIDs will be unique,
-- even after the OID counter wraps around.
-- We exclude non-system tables from the check by looking at nspname.
--
SELECT relname, nspname
FROM mdb_class c LEFT JOIN mdb_namespace n ON n.oid = relnamespace
WHERE relhasoids
    AND ((nspname ~ '^mdb_') IS NOT FALSE)
    AND NOT EXISTS (SELECT 1 FROM mdb_index i WHERE indrelid = c.oid
                    AND indkey[0] = -2 AND indnatts = 1
                    AND indisunique AND indimmediate);
 relname | nspname 
---------+---------
(0 rows)

